benchmarks/src/bench/json_multi_export.rs

```diff
@@ -52,24 +61,28 @@ impl Benchmark for JsonMultiExportBenchmark {
 
                 let ok: anyhow::Result<()> = Ok(());
                 ok
-            });
+            }));
         }
 
-        while let Some(result) = tasks.next().await {
-            result?;
+        for task in tasks {
+            task.await?;
         }
 
-        Ok(JsonMultiExportBenchmark { db, coll })
+        Ok(JsonMultiExportBenchmark {
+            uri: options.uri,
+            db,
+            coll,
+        })
     }
 
     async fn do_task(&self) -> Result<()> {
-        let mut tasks = FuturesUnordered::new();
+        let mut tasks = Vec::new();
 
         for i in 0..TOTAL_FILES {
-            let coll_ref = self.coll.clone();
+            let coll_ref = self.coll.clone_with_type::<Tweet>();
             let path = std::env::temp_dir();
 
-            tasks.push(async move {
+            tasks.push(crate::spawn(async move {
```

Comment:
these weren't actually running in parallel since it was the same task moving them all forward. This has been updated to spawn a new task for each worker, which greatly sped up this benchmark.